### PR TITLE
Add JUCE-based versions

### PR DESCRIPTION
**Pull request by getdunne**
You might already have this, but if not, here's a pull request to simplify pulling in these changes.
-shane

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,19 +6,15 @@
...
 	$(MAKE) -C common
+	$(MAKE) -C dpf/dgl ../build/libdgl-opengl.a
 
-dgl:
-ifeq ($(HAVE_DGL),true)
-	$(MAKE) -C dpf/dgl
-endif
-
-plugins: libs dgl
+plugins: libs
```
Comment (michaelwillis) [inline Makefile:17]: What is the rationale for editing the Makefile in the root directory? Does the `dgl` library conflict with something?

File: juce-plugins/DragonflyRoom/Source/DragonflyRoomProcessor.cpp [hunk 2]
```diff
@@ -116,40 +116,49 @@ typedef struct {
...
-    }
+   {
+     "Unify Custom", {       // dry, early, e. send, late, size,  width, delay, decay, diffuse, spin, wander, high cut, early damp, late damp, bass boost, boost lpf, low cut
+      {"Natural Wood",       { 85.0,  10.0,    20.0, 20.0, 15.0,  100.0,  15.0,  0.15,    80.0,  0.0,   20.0,    16000,       5200,    4000.0,       70.0,       850,  167.0 }},
+      {"Perc Room",          { 85.0,  15.0,    20.0, 13.0, 13.0,   90.0,  25.0,   0.3,    70.0,  0.4,   40.0,    16000,      16000,   10434.5,       20.0,       600,  110.0 }},
+      {"Glass Walls",        { 85.0,  10.0,    20.0, 20.0, 24.0,   90.0,  25.0, 0.327,    84.0,  0.4,   40.0,    15000,      14000,   12000.0,       60.0,       600,   60.0 }},
+      {"SuperLive",          { 80.0,  25.0,    60.0,  9.0,  8.3,   90.0,  25.0,  1.28,    25.0,  2.5,   80.0,    16000,      12000,    9000.0,       20.0,       500,   60.0 }},
+      {"Long Room",          { 85.0,   5.0,    20.0, 12.0, 11.94, 130.0, 100.0,   4.3,   100.0,  2.0,   40.0,     9755,      16000,   10000.0,       90.0,       479,   60.0 }}
+     }
+   },
```
Comment (michaelwillis) [inline juce-plugins/DragonflyRoom/Source/DragonflyRoomProcessor.cpp:127]: It's fun to see that you added some presets, I'll have to try these out.

**Review by michaelwillis — COMMENTED** [review]
Looks good! Just curious, are the JUCE plugins backward compatible with the DPF ones? Meaning, if somebody starts a project using the DPF plugins, can they seamlessly replace them with the JUCE plugins and expect the parameters to load properly and the plugins to sound the same? If not, that's ok, but we might consider consolidating on JUCE now that you've done all of this work, I just don't want to break existing projects.

**Comment by getdunne** [comment]
> Looks good! Just curious, are the JUCE plugins backward compatible with the DPF ones? Meaning, if somebody starts a project using the DPF plugins, can they seamlessly replace them with the JUCE plugins and expect the parameters to load properly and the plugins to sound the same? If not, that's ok, but we might consider consolidating on JUCE now that you've done all of this work, I just don't want to break existing projects.

Hi Michael. I just merged my "pluginguru" branch into the master branch today. It didn't occur to me that you might get any notification about it. I'll send you email to discuss further. For now, it's probably simplest to simply reject the pull request.

**Comment by JamesPeters667** [comment]
Please consider the issue below with JUCE-based plugins in Linux before making Dragonfly based on JUCE. This applies to Reaper and Ardour with JUCE-based Linux VST and LV2 as well:

https://forum.cockos.com/showthread.php?t=237954

I'm avoiding JUCE-based plugins as much as possible since it's a workflow killer if I can't start/stop playback in any way, if I don't "focus" a different window explicitly before doing it.

**Comment by getdunne** [comment]
Keystroke handling is a real bugaboo in JUCE. Most GUI widgets seem to grab input focus by default, and you have to add code to every single one, to tell it not to. I struggle with this all the time in my JUCE based projects. I'll add a note to my to-do list to look into this for the Dragonfly reverbs, but it's a very long list, so I can't say when I'll be able to get to it.

**Comment by JamesPeters667** [comment]
Thanks for that info, Shane. At least I now know I shouldn't blame Reaper or Ardour for this. :)

Do you know if the JUCE developers/maintainers know this? So many new Linux VST and also LV2 (Surge for instance) are using JUCE, and it seems to affect all of them. It's one of those things that makes Linux seem "irritating to use" compared to Windows as a result.

**Comment by getdunne** [comment]
> Do you know if the JUCE developers/maintainers know this?
I'm sure they do, but this is probably not close to the top of their to-do list either.
This is not so much a weakness of JUCE itself, as of their documentation/tutorials. What makes JUCE hard to use is that so much about it isn't written down.

**Pull request closed without merging**